### ISSUE TITLE
StateError when reusing undisposed PhoneController

### ISSUE THREAD

**Issue by JDVila** [cedvdb/phone_form_field#106]
Using the PhoneFormField in a bottomsheet with a PhoneController. When reopening the bottomsheet (using the same PhoneController), we get this error:
```
// _StreamControllerLifeCycle interface

  StreamSubscription<T> _subscribe(void onData(T data)?, Function? onError,
      void onDone()?, bool cancelOnError) {
    if (!_isInitialState) {
      throw StateError("Stream has already been listened to.");
    }
```

Specifically, the line `throw StateError("Stream has already been listened to.");` gets triggered. It looks like once a value is pulled once, the stream no longer emits values, even if the controlled has yet to be disposed. Is this issue known? What is the current workaround? Thank you for your help!

**Comment by cedvdb** [comment]
This issue is not known and needs a repro from you side

**Comment by JDVila** [comment]
Hello, and thank you for your reply!

This issue occurs because I set the `value` of the `PhoneController` multiple times (which is expected behavior with other controllers, like `TextEditingController`, for example). 

I instantiate the controller like so:

```
@override
void initState() {
    phoneController = PhoneController(initialPhoneNumber);
    ....
```

I dispose of it when it's no longer needed:

```
  @override
  void dispose() {
    super.dispose();
    phoneController.dispose();
    ....
```

I assign it to a `PhoneFormField` widget:

```
PhoneFormField(
                        autovalidateMode: AutovalidateMode.onUserInteraction,
                        controller: phoneController,
                        validator: PhoneValidator.validMobile(),
                        keyboardType: TextInputType.phone,
                        decoration: const InputDecoration(
                          border: OutlineInputBorder(),
                          labelText: 'Phone',
                          hintText: 'eg. 1-212-555-1212',
                        ),
                      ),
```

And I clear it's value when I want to reuse it:

```
phoneController.reset();
```

However, if I want to re-open the bottomsheet where this `PhoneFormField` is used, I get the error.

Here is the stacktrace:

```
════════ Exception caught by widgets library ═══════════════════════════════════
The following StateError was thrown building _FormScope:
Bad state: Stream has already been listened to.

The relevant error-causing widget was
Form-[LabeledGlobalKey<FormState>#a2031]
When the exception was thrown, this was the stack
#3      _PhoneFormFieldState.initState
#4      StatefulElement._firstBuild
#5      ComponentElement.mount
...     Normal element mounting (4 frames)
#9      Element.inflateWidget
#10     MultiChildRenderObjectElement.inflateWidget
#11     MultiChildRenderObjectElement.mount
...     Normal element mounting (384 frames)
#395    Element.inflateWidget
#396    MultiChildRenderObjectElement.inflateWidget
#397    Element.updateChild
#398    RenderObjectElement.updateChildren
#399    MultiChildRenderObjectElement.update
#400    Element.updateChild
#401    ComponentElement.performRebuild
#402    StatefulElement.performRebuild
#403    Element.rebuild
#404    StatefulElement.update
#405    Element.updateChild
#406    ComponentElement.performRebuild
#407    Element.rebuild
#408    ProxyElement.update
#409    Element.updateChild
#410    ComponentElement.performRebuild
#411    Element.rebuild
#412    ProxyElement.update
#413    _InheritedNotifierElement.update
#414    Element.updateChild
#415    SingleChildRenderObjectElement.update
#416    Element.updateChild
#417    ComponentElement.performRebuild
#418    StatefulElement.performRebuild
#419    Element.rebuild
#420    StatefulElement.update
#421    Element.updateChild
#422    SingleChildRenderObjectElement.update
#423    Element.updateChild
#424    SingleChildRenderObjectElement.update
#425    Element.updateChild
#426    ComponentElement.performRebuild
#427    Element.rebuild
#428    ProxyElement.update
#429    Element.updateChild
#430    ComponentElement.performRebuild
#431    StatefulElement.performRebuild
#432    Element.rebuild
#433    BuildOwner.buildScope
#434    WidgetsBinding.drawFrame
#435    RendererBinding._handlePersistentFrameCallback
#436    SchedulerBinding._invokeFrameCallback
#437    SchedulerBinding.handleDrawFrame
#438    SchedulerBinding._handleDrawFrame
#442    _invoke (dart:ui/hooks.dart:151:10)
#443    PlatformDispatcher._drawFrame (dart:ui/platform_dispatcher.dart:308:5)
#444    _drawFrame (dart:ui/hooks.dart:115:31)
(elided 6 frames from dart:async)
════════════════════════════════════════════════════════════════════════════════
```

**Comment by cedvdb** [comment]
Do you have the line number for  `#3      _PhoneFormFieldState.initState`.

Could you change the line I provided above to see if that fixes the issue ?

**Comment by JDVila** [comment]
Thank you for your reply! The line of code on line #3 is on lines 23/24 of `phone_form_field_state.dart`:

```
_selectionSubscription = _controller.selectionRequest$
        .listen((event) => _childController.selectNationalNumber());
```

**Comment by cedvdb** [comment]
Can you try the latest dev branch. I just pushed a fix.

**Comment by JDVila** [comment]
Thank you so much! It works when using the git url dependency in the `pubspec.yaml` file option you suggested! What was the issue that needed to be updated?

**Comment by cedvdb** [comment]
I needed to make the StreamController broadcast, as it was being listened to multiple times when you were repassing the controller.

In other words, in dart, `StreamController().stream` can be listened to only once. Broadcasting changes that.

**Comment by cedvdb** [comment]
fixed in 5.0.3 released on pub dev.

Thank you.
